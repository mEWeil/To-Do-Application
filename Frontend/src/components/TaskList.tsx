import { useState, useEffect } from 'react';
import axios from 'axios';
import CsrfToken from './CsrfToken';

export interface ITask {
  id: number;
  title: string;
  status: boolean;
}

export async function getTasks(): Promise<ITask[]> {
  const resp = await axios.get('list_tasks');
  return resp.data.tasks;
}

export default function TaskList() {
  const [tasks, setTasks] = useState<ITask[]>([]);

  useEffect(() => {
    const getResp = async () => {
      const resp = await getTasks();
      setTasks(resp);
    };
    getResp();
  }, []);

  CsrfToken();

  return (
    <div>
      <div>
        <h1>Pending</h1>
        <ul>
          {tasks
            .filter((task) => task.status === false)
            .map((task) => {
              return (
                <li key={task.id}>
                  `{task.id}) {task.title}`
                </li>
              );
            })}
        </ul>
      </div>
      <div>
        <h1>Completed</h1>
        <ul>
          {tasks
            .filter((task) => task.status === true)
            .map((task) => {
              return (
                <li key={task.id}>
                  `{task.id}) {task.title}`
                </li>
              );
            })}
        </ul>
      </div>
    </div>
  );
}
