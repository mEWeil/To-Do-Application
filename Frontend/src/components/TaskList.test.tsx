import { describe, expect, it, vi, Mocked } from 'vitest';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import axios from 'axios';
import TaskList, { getTasks } from './TaskList';

vi.mock('axios');

describe('TaskList.tsx', () => {
  describe('getTasks()', () => {
    it('return an array of tasks', async () => {
      const mockedAxios = axios as Mocked<typeof axios>;
      mockedAxios.get.mockResolvedValue({ data: { tasks: [] } });
      const tasks = await getTasks();
      expect(tasks).toStrictEqual([]);
    });
  });

  describe('taskList.tsx', () => {
    it('will create and match snapshot', () => {
      const myTaskList = TestRenderer.create(<TaskList />);
      expect(myTaskList).toMatchSnapshot();
    });
  });
});
