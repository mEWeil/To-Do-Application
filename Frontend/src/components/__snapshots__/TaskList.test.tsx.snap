// Vitest Snapshot v1

exports[`TaskList.tsx > taskList.tsx > will create and match snapshot 1`] = `
<div>
  <div>
    <h1>
      Pending
    </h1>
    <ul />
  </div>
  <div>
    <h1>
      Completed
    </h1>
    <ul />
  </div>
</div>
`;
